lib/membrane/video_compositor/handler.ex

```diff
@@ -2,90 +2,217 @@ defmodule Membrane.VideoCompositor.Handler do
   @moduledoc """
   Module defining behaviour of handlers.
 
-  Implementing handler allows to provide custom implementation and
+  Implementing a handler allows to provide custom implementation and
   react to various events, among others by setting a new scene and/or
   the inner custom state.
+
+  ## Examples
+
+  ### Simple video conference room
+  Video Compositor is used in video conference app like the videoroom.
+  Composition is determined by number of users.
+  This is how handler module might look like:
+
+  ```
+  defmodule ExampleHandler do
+    @moduledoc false
+
+    @behaviour Membrane.VideoCompositor.Handler
+
+    alias Membrane.VideoCompositor.Scene
+
+    @impl true
+    def handle_init(_init_options) do
+      %{user_count_to_scene: get_users_scene_mapping()}
+    end
+
+    @impl true
+    def handle_inputs_change(inputs, _ctx, state) do
+      {Map.get(state.user_count_to_scene, map_size(inputs), default_scene), state}
+    end
+
+    @impl true
+    def handle_info(msg, _ctx, _state) do
+      raise "Unsupported message \#{msg}"
+    end
+
+    @impl true
+    def handle_scene_expire(_expired_scene, _ctx, _state) do
+      raise "This shouldn't happen"
+    end
```

Comment:
- You can delete it from this example, IMO it only confuses the user
- We should create a separate task to add default callback implementation to avoid this